src/modules/Modal/ModalActions.js

```diff
@@ -1,23 +1,32 @@
 import cx from 'classnames'
+import _ from 'lodash'
 import React, { PropTypes } from 'react'
 
 import {
+  createShorthandFactory,
   customPropTypes,
   getElementType,
   getUnhandledProps,
   META,
 } from '../../lib'
+import Button from '../../elements/Button'
 
 /**
  * A modal can contain a row of actions.
  */
 function ModalActions(props) {
-  const { children, className } = props
+  const { actions, children, className } = props
   const classes = cx('actions', className)
   const rest = getUnhandledProps(ModalActions, props)
   const ElementType = getElementType(ModalActions, props)
 
-  return <ElementType {...rest} className={classes}>{children}</ElementType>
+  if (!_.isNil(children)) return <ElementType {...rest} className={classes}>{children}</ElementType>
+
+  return (
+    <ElementType {...rest} className={classes}>
+      {actions.map((action) => Button.create(action, true))}
```

Comment:
Ah, this is a new use case.  We currently don't have a way to enable keys for a create method during runtime like this.  Instead, the create method itself is defined as supporting keys or not.  The button currently does not support keys.

The second param here is actually `defaultProps`.  In order to support keys, we need to pass `true` as the 3rd param to `createShorthandFactory` for the Button's `create` method:

```diff
-Button.create = createShorthandFactory(Button, value => ({ content: value }))
+Button.create = createShorthandFactory(Button, value => ({ content: value }), true)
```

We have a PR up for factory updates.  I'll enable keys for all factories in light of this issue.  We want users to be able to use the `create` method themselves as well, and, they may need keys.